sqflite/android/src/main/java/com/tekartik/sqflite/Database.java

```diff
@@ -391,25 +390,14 @@ private boolean executeOrError(Operation operation) {
         if (LogLevel.hasSqlLevel(logLevel)) {
             Log.d(TAG, getThreadLogPrefix() + command);
         }
-        boolean operationInTransaction = Boolean.TRUE.equals(operation.getInTransactionChange());
-
+        Boolean operationInTransaction = operation.getInTransactionChange();
         try {
             getWritableDatabase().execSQL(command.getSql(), command.getSqlArguments());
-
-            // Success handle inTransaction change
-            if (operationInTransaction) {
-                inTransaction = true;
-            }
+            enterOrLeaveInTransaction(operationInTransaction);
             return true;
         } catch (Exception exception) {
             handleException(exception, operation);
             return false;
-        } finally {
```

Comment:
The logic here is important to keep if the command (likely BEGIN TRANSACTION) fails, we should not remain in a transaction mode

Reply:
note to myself: add a test on that to test maybe we something like BEGIN DUMMY (that should be parsed internally as a transaction change but should fail).

Reply:
Reading again I don't think it should be in the finally section, I need to spend more time on this.

Reply:
If something fail, shall we keep current state so next retry (if any) will be run in the same context.